module/VuFind/src/VuFind/Sitemap/Generator.php

```diff
@@ -276,14 +317,49 @@ public function generate()
         }
 
         // Set-up Sitemap Index
-        $this->buildIndex($currentPage - 1);
+        $this->buildIndex($currentPage - 1, $additionalSitemaps);
 
         // Display total elapsed time in verbose mode:
         $this->verboseMsg(
             'Elapsed time (in seconds): ' . round($this->getTime() - $startTime)
         );
     }
 
+    /**
+     * Generate sitemaps with any enabled plugins
+     *
+     * @return array
+     */
+    protected function generateWithPlugins(): array
+    {
+        $additionalSitemaps = [];
+        if ($plugins = $this->config->Sitemap->plugins) {
```

Comment:
Does this behave appropriately if the setting is missing entirely? Just want to be sure there's no risk of notices.

Reply:
Yes. As long as $this->config->Sitemap is \Laminas\Config\Config, it works and just returns null. The constructor is already doing that.

Reply:
Actually, if the entire [Sitemap] section is missing, this does cause a notice, as I discovered when beginning to build a Generator test class. I addressed this in 4b4499994b, along with using a class property in place of a duplicate config lookup. Please let me know if you have any concerns about this!